sdk/azidentity/azure_cli_credential.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// Enables authentication to Azure Active Directory using Azure CLI to generated an access token.
+type AzureCLICredential struct {
```

Comment:
Rename this to DeveloperCredential (we may change this later)

Reply:
Don't rename this right now. Let's get everything checked in as `AzureCliCredential` and then we can lay `DeveloperCredential` on top of this afterwards.